source/crud/bulk-write.md

```diff
@@ -0,0 +1,724 @@
+- Introduce separate `SummaryBulkWriteResult` and `VerboseBulkWriteResult` types.
+  `VerboseBulkWriteResult` MUST have all of the required fields defined on `BulkWriteResult` above.
+  `SummaryBulkWriteResult` MUST have all of the required fields defined on `BulkWriteResult` above
+  except `insertResults`, `updateResults`, and `deleteResults`.
+
+#### Individual Results
+
+The `InsertOneResult`, `UpdateResult`, and `DeleteResult` classes are the same as or similar to
+types of the same name defined in the [CRUD specification](crud.md). Drivers MUST redefine these
+classes if their existing result classes deviate from the definitions in this specification (e.g.
+if they contain acknowledgement information, which is not applicable for individual bulk write
+operations). Drivers MAY reuse their existing types for these classes if they match the ones
+defined here exactly.
+
+### Exception
+
+```typescript
+class BulkWriteException {
+    /**
+     * A top-level error that occurred when attempting to communicate with the server or execute
+     * the bulk write. This value may not be populated if the exception was thrown due to errors
+     * occurring on individual writes.
+     */
+    error: Optional<Error>;
+
+    /**
+     * Write concern errors that occurred while executing the bulk write. This list may have
+     * multiple items if more than one server command was required to execute the bulk write.
+     */
+    writeConcernErrors: WriteConcernError[];
+
+    /**
+     * Errors that occurred during the execution of individual write operations. This map will
+     * contain at most one entry if the bulk write was ordered.
+     */
+    writeErrors: Map<Int64, WriteError>;
+
+    /**
+     * The results of any successful operations that were performed before the error was
+     * encountered.
+     */
+    partialResult: Optional<BulkWriteResult>;
+}
+```
+
+## Building a `bulkWrite` Command
+
+The `bulkWrite` server command has the following format:
+
+```json
+{
+    "bulkWrite": 1,
+    "ops": <Array>,
+    "nsInfo": <Array>,
+    "errorsOnly": <Boolean>,
+    "ordered": <Boolean>,
+    "bypassDocumentValidation": <Boolean>,
+    "comment": <Bson>,
+    "let": <Document>,
+    ...additional operation-agnostic fields
+}
+```
+
+Drivers SHOULD use document sequences ([`OP_MSG`](../message/OP_MSG.rst) payload type 1) for the
+`ops` and `nsInfo` fields. Drivers MUST NOT use document sequences when auto-encryption is enabled.
+
+The `bulkWrite` command is executed on the "admin" database.
+
+### Operations
+
+The `ops` field is a list of write operation documents. The first entry in each document has
+the name of the operation (i.e. "insert", "update", or "delete") as its key and the index in the
+`nsInfo` array of the namespace on which the operation should be performed as its value. The
+documents have the following format:
+
+#### Insert
+
+```json
+{
+    "insert": <Int32>,
```

Comment:
Is the server flexible about accepting an Int32 or Int64 here, or is Int32 absolutely required? I do realize how absurd an Int64 would be in this context.